subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyResolveTimeoutIntegrationTest.groovy

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve
+
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.test.fixtures.keystore.TestKeyStore
+import org.gradle.test.fixtures.maven.MavenFileRepository
+import org.gradle.test.fixtures.maven.MavenModule
+import org.gradle.test.fixtures.maven.MavenRepository
+import org.gradle.test.fixtures.server.http.HttpResource
+import org.gradle.test.fixtures.server.http.MavenHttpModule
+import org.gradle.test.fixtures.server.http.MavenHttpRepository
+import spock.lang.Unroll
+
+import static org.gradle.internal.resource.transport.http.JavaSystemPropertiesHttpTimeoutSettings.SOCKET_TIMEOUT_SYSTEM_PROPERTY
+
+class DependencyResolveTimeoutIntegrationTest extends AbstractHttpDependencyResolutionTest {
+
+    private static final String GROUP_ID = 'group'
+    private static final String VERSION = '1.0'
+    TestFile downloadedLibsDir
+    MavenHttpModule moduleA
+
+    def setup() {
+        moduleA = publishMavenModule(mavenHttpRepo, 'a')
+        downloadedLibsDir = file('build/libs')
+        executer.withArgument("-D${SOCKET_TIMEOUT_SYSTEM_PROPERTY}=1000")
+    }
+
+    void blockingForProtocol(String protocol, HttpResource... resources) {
+        if (protocol == 'http') {
+            resources.each { it.expectGetBlocking() }
+        } else if (protocol == 'https') {
+            // https://issues.apache.org/jira/browse/HTTPCLIENT-1478
+            def keyStore = TestKeyStore.init(temporaryFolder.file('ssl-keystore'))
+            keyStore.enableSslWithServerCert(server)
+            keyStore.configureServerCert(executer)
+            server.expectSslHandshakeBlocking()
+        } else {
+            assert false
```

Comment:
:x: It would be better if this would throw an exception with a descriptive message if the protocol is other than `http` or `https`.

Reply:
Done.